examples/v1beta1/taskruns/array-default.yaml

```diff
@@ -0,0 +1,31 @@
+apiVersion: tekton.dev/v1beta1
+kind: TaskRun
+metadata:
+  generateName: array-with-default-
+spec:
+  params:
+    - name: flags
+      value:
+        - "foo"
+        - "bar"
+  taskSpec:
+    params:
+      - name: flags
+        type: array
+      - name: flags-default
+        type: array
+        default:
+        - "foo-default"
+        - "bar-default"
+      - name: singleVal
+        type: string
+        default: "baz"
+    steps:
+      - name: build
+        image: bash:3.2
```

Comment:
Nit: IMO, adding a comment that documents the fully-resolved `echo` that will be executed by the task post-parameter-resolution will clarify the example.